src/VSCodeExtension/src/yeoman-generator.ts

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import * as vscode from 'vscode';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import yo = require("yeoman-generator");
+import yosay = require("yosay");
+
+export class QSharpGenerator extends yo {
+
+    constructor(args : any, opts : any) {
+        super(args, opts);
+        console.log(
+            yosay("Welcome to the Q# generator!")
+        );
+
+        this.sourceRoot(path.join(this.options.extensionPath, "templates"));
+    }
+
+    prompting() {
+        let done = this.async();
+
+        // This dictionary maps the public description of the project type to the name
+        // of the folder with the corresponding template files.
+        const projectTypes: {[key: string]: string} = {
+            "Standalone console application": "application",
+            "Quantum library": "library",
+            "Unit testing project": "unittest"
+        };
+
+        vscode.window.showQuickPick(
```

Comment:
As discussed, I think it would be a good future addition to refactor this to use standard Yeoman prompting APIs, and to move showQuickPick into the adapter. That is likely beyond the scope of this PR, though; just wanted to flag it for the future.

Reply:
Sounds good, I'll open a task to track this for a future release.